GoogleDataTransport/GDTCORLibrary/GDTCOREventMetricsCounter.m

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "GoogleDataTransport/GDTCORLibrary/Private/GDTCOREventMetricsCounter.h"
+
+#import "GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCOREvent.h"
+
+typedef NSDictionary<NSNumber *, NSNumber *> GDTCORDroppedEventCounter;
+
+@interface GDTCOREventMetricsCounter ()
+
+// TODO(ncooke3): Document.
+@property(nonatomic)
+    NSDictionary<NSString *, GDTCORDroppedEventCounter *> *droppedEventCounterByMappingID;
+
+@end
+
+@implementation GDTCOREventMetricsCounter
+
++ (instancetype)counterWithEvents:(NSArray<GDTCOREvent *> *)events
+                 droppedForReason:(GDTCOREventDropReason)reason {
+  NSMutableDictionary<NSString *, GDTCORDroppedEventCounter *> *eventCounterByMappingID =
+      [NSMutableDictionary dictionary];
+
+  for (GDTCOREvent *event in events) {
+    // TODO(ncooke3): Should I use an autorelease pool?
```

Comment:
Why do you ask?  How big is the event set likely to get?  How is metrics event processing different than other events processing?

Reply:
I asked because I thought that if the event set was quite large, repeatedly allocating a dictionary in this scope could be costly. 

I was going to profile it to be decide later when i get to integration testing. I suspect it would be fine to not use one.